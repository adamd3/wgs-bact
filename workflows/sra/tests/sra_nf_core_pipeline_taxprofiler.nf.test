nextflow_workflow {

    name "Test workflow: sra/main.nf"
    script "../main.nf"
    workflow "SRA"
    tag "workflows"
    tag "workflows_sra"
    tag "multiqc_mappings_config"
    tag "sra_fastq_ftp"
    tag "sra_ids_to_runinfo"
    tag "sra_merge_samplesheet"
    tag "sra_runinfo_to_ftp"
    tag "sra_to_samplesheet"
    tag "sra_nf_core_pipeline_taxprofiler"

    test("Parameters: --nf_core_pipeline taxprofiler") {

        when {
            workflow {
                """
                input[0] = Channel.from('ERR1160846', 'GSE214215', 'SRR12848126')
                """
            }
            params {
                outdir = "results"
                nf_core_pipeline = "taxprofiler"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
