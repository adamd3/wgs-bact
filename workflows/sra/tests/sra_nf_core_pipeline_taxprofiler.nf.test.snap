{
    "Parameters: --nf_core_pipeline taxprofiler": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "samplesheet.csv:md5,fe3c7ba9e7b7debe07291d6125f9edc7"
                ],
                "2": [
                    "id_mappings.csv:md5,f7c0bd1713fe14917b4298e252312296"
                ],
                "3": [
                    [
                        "multiqc_config.yml:md5,6d4c3e5137704358474330207f5f2b5c",
                        "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9"
                    ]
                ],
                "4": [
                    "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9",
                    "versions.yml:md5,45e4a369df2d3af023b8622aa3f6d5a8",
                    "versions.yml:md5,98a3ee9aa3bcc3b080247bccde481bbb",
                    "versions.yml:md5,9b17045ca8bdc272cb3f9d349a81d206",
                    "versions.yml:md5,b52279f7d6b891a6523d9321f3f85b47"
                ],
                "fastq": [
                    
                ],
                "mappings": [
                    "id_mappings.csv:md5,f7c0bd1713fe14917b4298e252312296"
                ],
                "sample_mappings": [
                    [
                        "multiqc_config.yml:md5,6d4c3e5137704358474330207f5f2b5c",
                        "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9"
                    ]
                ],
                "samplesheet": [
                    "samplesheet.csv:md5,fe3c7ba9e7b7debe07291d6125f9edc7"
                ],
                "versions": [
                    "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9",
                    "versions.yml:md5,45e4a369df2d3af023b8622aa3f6d5a8",
                    "versions.yml:md5,98a3ee9aa3bcc3b080247bccde481bbb",
                    "versions.yml:md5,9b17045ca8bdc272cb3f9d349a81d206",
                    "versions.yml:md5,b52279f7d6b891a6523d9321f3f85b47"
                ]
            }
        ],
        "timestamp": "2024-01-05T17:47:55.191319"
    }
}